### PR TITLE
feat: create user POST endpoint (user registration)

### PR DESCRIPTION
**Pull request by MjxOro**
# Summary
First back-end PR, I learned a lot and had a lot of fun writing the test for this PR. I created a user model which would store `email`(String), `username`(String), `token`(String),  `hash` (password)(String), `bio`(String), `image`(String).  I installed JSON web token for `token` value and bcrypt to help salt and hash the password stored in the database as `hash`. To grab all these values and store in the database, I created a controller which takes the POST application/JSON request in the body that validates and then stores them in the database. I also wrote a basic test that would send a post request with registration data and expect an object back with `email`,`token`,`username`,`bio`,`image` as the properties. after the request is done the test will remove the test values from the database.

#Screenshot of the test
Screenshot 2022-10-04 at 12 05 48 PM

### CODE REVIEW

File: src/controllers/users.js
Comment (YoussefLaunchUp) [inline src/controllers/users.js]: Duplicated code here, Lets change validateBody to also check if body exists. It can also be used to return what the error is so a more detailed error is sent to the user.

File: src/models/users.js
Comment (YoussefLaunchUp) [inline src/models/users.js:7]: Email should be set as the primary key for users

File: src/utils/index.js
Comment (YoussefLaunchUp) [inline src/utils/index.js]: Change as commented on the controller.

Comment (YoussefLaunchUp) [reply src/utils/index.js]: validateBody also needs unit tests.

File: test/integration/register-user.test.js
Comment (YoussefLaunchUp) [inline test/integration/register-user.test.js]: Since this is the same for the integration tests, lets separate it out so it can be reused in both.

**Review by YoussefLaunchUp — CHANGES_REQUESTED** [review]
Great work on first Backend PR! Some changes needed before approval.

File: src/models/users.js
Comment (MjxOro) [reply src/models/users.js:7]: refactored the model to have email as primary key

File: src/controllers/users.js
Comment (MjxOro) [reply src/controllers/users.js]: done, removed duplicate code, and gave more detailed error message to user

File: src/utils/index.js
Comment (MjxOro) [reply src/utils/index.js]: to clarify, I should comment the example /api/ endpoint?
 ```javascript
router.get("/", (req, res) => {
  res.send(emptyEndpointResponse());
});
```

Comment (MjxOro) [reply src/utils/index.js]: I cleaned up by removing example code

File: test/integration/register-user.test.js
Comment (MjxOro) [reply test/integration/register-user.test.js]: I created the import setup on `src/utils/integrationTestSetup.js` to reuse base setup.

File: test/unit/req-body-verification.test.js
Comment (YoussefLaunchUp) [inline test/unit/req-body-verification.test.js:45]: Great work on the unit tests

File: src/utils/index.js
Comment (YoussefLaunchUp) [inline src/utils/index.js]: Please rename the file from index.js to something more descriptive. 
Also the function needs to validate if emails are correct as well.

**Review by YoussefLaunchUp — CHANGES_REQUESTED** [review]

File: src/utils/index.js
Comment (MjxOro) [reply src/utils/index.js]: I renamed index.js to validators.js

**Review by YoussefLaunchUp — APPROVED** [review]
Great job!

**Pull request merged**
